### PR TITLE
API docs: only index top-level symbols

### PR DESCRIPTION
**Pull request by slimsag**
This adds a failing test showing that non-top-level symbols are currently indexed by API docs, and then fixes the issue. Using an AST visitor was definitely incorrect here, not sure how I missed it :)

Fixes sourcegraph/sourcegraph#22030

### CODE REVIEW

File: buildkite.yml [hunk 1]
```diff
@@ -1,3 +1,5 @@
 steps:
+  - label: ':go:'
+    command: find internal/testdata -type f -name '*.go' -exec go fmt {} \;
```
Comment (efritz) [inline buildkite.yml:3]: Stacked PR?

Comment (slimsag) [reply buildkite.yml:3]: yeah, rebasing now.

File: internal/indexer/documentation.go [hunk 2]
```diff
@@ -299,13 +299,9 @@ func (d *docsIndexer) indexFile(p *packages.Package, f *ast.File, isTestFile boo
...
 	result.pkgDocsMarkdown = godocToMarkdown(f.Doc.Text())
 
-	// Visit each AST node, collecting the bits we want to emit documentation for.
-	var (
-		firstErr error
-		visitor  ast.Visitor
-	)
-	visitor = visitorFunc(func(n ast.Node) (w ast.Visitor) {
-		switch node := n.(type) {
+	// Collect each top-level declaration.
```
Comment (efritz) [inline internal/indexer/documentation.go:302]: Oh this is probably much faster as well than walking the entire AST.

**Review by efritz — APPROVED** [review]

File: internal/indexer/documentation.go (hunk 2 above)
Comment (slimsag) [reply internal/indexer/documentation.go:302]: yeah :D

**Pull request merged**